lib/datastore/rocksdb.go

```diff
@@ -0,0 +1,116 @@
+package datastore
+
+import (
+	"github.com/ByteStorage/FlyDB/config"
+	"github.com/ByteStorage/FlyDB/lib/encoding"
+	"github.com/hashicorp/raft"
+	"github.com/tecbot/gorocksdb"
+	"sync"
+)
+
+// RocksDbStore is a struct that implements the raft.LogStore interface
+// It uses BoltDB as the underlying storage and a read-write mutex for concurrency control
+type RocksDbStore struct {
+	mux  sync.RWMutex
+	conn *gorocksdb.DB
+}
+
+// NewLogRocksDbStorage is a function that creates a new RocksDB store
+// It takes a configuration map as input and returns a raft.LogStore and an error
+func NewLogRocksDbStorage(conf config.Options) (raft.LogStore, error) {
+	filename := conf.DirPath
+	options := gorocksdb.NewDefaultOptions()
```

Comment:
It seems that a method that supports user-defined rocksdb configuration can be added in the later stage, but there is currently no rush to implement it.